### DISCUSSION TITLE
Crazyswarm position streaming command

### DISCUSSION THREAD

**Discussion by zafer4481** [bitcraze/discussions#73] [Q&A]
Hi I am writing a high level controller to control multiple drones in formation. For this I use the Crazyswarm environment and already can form desired shapes and track trajectories for a swarm of drones. I use the cmdPosition(pos, yaw=0) command, see: https://crazyswarm.readthedocs.io/en/latest/api.html

At the moment I stream setpoints in 10Hz which they claim is a conservative minimum. My question is how high I can go with this streaming frequency? Theoratically the higher the better, but is 100Hz for example realistic? Will the setpoints really be streamed with 100Hz if I code it like that, or is there a limitting factor?

**Comment by krichardsson** [comment]
You might get a better answer to this question in the [Crazyswarm discussions](https://github.com/USC-ACTLab/crazyswarm/discussions) but I'll give it a go.
Generally the bandwidth of a Crazy radio is shared by the Crazyflies that is communicating over that radio. This means that the the frequency you can use to send packets to the Crazyflie depends on how many Crazflies and radios you are using. Another limiting factor is what other information that is communicated on the link, if you are using a mocap system for instance, the position information is also sent to the Crazyflie on the same link.

A Crazyradio can send around 800 small packets to one Crazyflie per second, if the packets are larger the rate may go down to around 400/s.
